services/src/datasets/external/gbif.rs

```diff
@@ -0,0 +1,2467 @@
+                        FeatureData::NullableText(vec![None, None]),
+                    ),
+                    (
+                        "verbatimscientificname".to_string(),
+                        FeatureData::NullableText(vec![
+                            Some("Bubo bubo (Linnaeus, 1758)".to_string()),
+                            Some("Bubo bubo (Linnaeus, 1758)".to_string()),
+                        ]),
+                    ),
+                    (
+                        "verbatimscientificnameauthorship".to_string(),
+                        FeatureData::NullableText(vec![
+                            Some("(Linnaeus, 1758)".to_string()),
+                            Some("(Linnaeus, 1758)".to_string()),
+                        ]),
+                    ),
+                ]
+                .iter()
+                .cloned()
+                .collect(),
+            )
+            .unwrap();
+
+            if result != &expected {
+                return Err(format!("{result:?} != {expected:?}"));
+            }
+
+            Ok(())
+        }
+
+        let db_config = config::get_config_element::<config::Postgres>().unwrap();
+
+        let test_schema = create_test_data(&db_config).await;
+
+        let result = test(&db_config, &test_schema).await;
+
+        cleanup_test_data(&db_config, test_schema).await;
+
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn it_cites() {
+        async fn test(db_config: &config::Postgres, test_schema: &str) -> Result<(), String> {
+            let provider = Box::new(GbifDataProviderDefinition {
+                name: "GBIF".to_string(),
+                db_config: DatabaseConnectionConfig {
+                    host: db_config.host.clone(),
+                    port: db_config.port,
+                    database: db_config.database.clone(),
+                    schema: test_schema.to_owned(),
+                    user: db_config.user.clone(),
+                    password: db_config.password.clone(),
+                },
+            })
+            .initialize()
+            .await
+            .map_err(|e| e.to_string())?;
+
+            let dataset = DataId::External(ExternalDataId {
+                provider_id: GBIF_PROVIDER_ID,
+                layer_id: LayerId("species/Bubo bubo".to_owned()),
+            });
+
+            let result = provider
+                .provenance(&dataset)
+                .await
+                .map_err(|e| e.to_string())?;
+
+            let expected = ProvenanceOutput {
+                data: DataId::External(ExternalDataId {
+                    provider_id: GBIF_PROVIDER_ID,
+                    layer_id: LayerId("species/Bubo bubo".to_owned()),
+                }),
+                provenance: Some(Provenance {
+                    citation: "GBIF.org (01 January 2023) GBIF Occurrence Download https://doi.org/10.15468/dl.yta4s7".to_owned(),
+                    license: "CC BY-NC 4.0".to_owned(),
```

Comment:
also here, please no NC in the codebase

Reply:
I've seen, there is a lot of CC0 data at gbif.org

Reply:
Test data is now CC0.